src/Veldrid/MTL/MTLGraphicsDevice.cs

```diff
@@ -518,69 +528,12 @@ protected override void UnmapCore(MappableResource resource, uint subresource)
         {
         }
 
-        public override bool WaitForFence(Fence fence, ulong nanosecondTimeout)
-        {
-            return Util.AssertSubtype<Fence, MTLFence>(fence).Wait(nanosecondTimeout);
-        }
+        public override bool WaitForFence(Fence fence, ulong nanosecondTimeout) =>
+            throw new NotImplementedException("Fences are now handled by GPU, this requires establishing a notification listener for the underlying event and waiting for a signal.");
```

Comment:
This doesn't read well to me. Are you suggesting a consumer has to do this themselves using the proposed "notification listener"?

Reply:
meant it more of a note for anyone who wants to implement, I can change it to a todo.

Reply:
The exception should be understandable to someone calling the function and expecting it to work.

If leaving a note about what is required to implement, it should be inline commented.

Reply:
I remember that's the proper usage of `NotImplementedException`s, but sure I'll change it.